src/python/pants/python/python_setup.py

```diff
@@ -110,11 +111,11 @@ def register_options(cls, register):
         register(
             "--resolver-jobs",
             type=int,
-            default=2,
+            default=multiprocessing.cpu_count() // 2,
             advanced=True,
             help=(
                 "The maximum number of concurrent jobs to build wheels with. Because Pants "
-                "can run multiple subprocesses in parallel, the total parallelism will be "
+                "can run multiple subprocesses in parallel, the maximum total parallelism will be "
                 "`--process-execution-{local,remote}-parallelism x --python-setup-resolver-jobs`. "
                 "Setting this option higher may result in better parallelism, but, if set too "
                 "high, may result in starvation and Out of Memory errors."
```

Comment:
It didn't seem worth it to mention the edge cases where you want this low vs. high. Way too much detail.